server/src/main/java/com/google/fhir/gateway/AllowedQueriesChecker.java

```diff
@@ -61,7 +60,18 @@ class AllowedQueriesChecker implements AccessChecker {
     }
   }
 
-  @Override
+  public AccessDecision checkUnAuthenticatedAccess(RequestDetailsReader requestDetails) {
+    if (config == null) {
+      return NoOpAccessDecision.accessDenied();
+    }
+    for (AllowedQueryEntry entry : config.entries) {
+      if (entry.isAllowUnAuthenticatedRequests() && requestMatches(requestDetails, entry)) {
```

Comment:
The amount of code duplication is very less. I thought it is better to keep the logic inline rather that extracting to a common function. 
Let me know if you strongly feel otherwise.